runtime-config.md

```diff
@@ -12,27 +12,28 @@ Presently there are `Prestart`, `Poststart` and `Poststop`.
 Hooks allow one to run code before/after various lifecycle events of the container.
 Hooks MUST be called in the listed order.
 The state of the container is passed to the hooks over stdin, so the hooks could get the information they need to do their work.
+All hooks execute in the host environment (e.g. the same namespace, cgroups, etc. that apply to the host process).
 
 Hook paths are absolute and are executed from the host's filesystem.
 
 ### Pre-start
 
-The pre-start hooks are called after the container process is spawned, but before the user supplied command is executed.
+The pre-start hooks are called [after the container process is spawned, but before the user supplied command is executed](runtime.md#typical-lifecycle).
```

Comment:
On Mon, Oct 12, 2015 at 12:47:00PM -0700, Vish Kannan wrote:
> > -The pre-start hooks are called after the container process is spawned, but before the user supplied command is executed.
> > +The pre-start hooks are called [after the container process is spawned, but before the user supplied command is executed](runtime.md#typical-lifecycle).
> 
> Do we need to mention `container process` here? These hooks are
> invoked once the majority of the container runtime has been setup,
> but before the container's init process has been spawned. We need to
> clearly specify what sub-section of the complete container
> environment is available for these hooks.

The container is complete when the pre-start hooks run.  The container
process has been spawned (it's just running runtime-supplied code).
The only change that happens *after* the pre-start hooks is that that
container process execs the user-supplied process, swapping in the
user-selected code.  For a bit of runC-side docs on this, see the
synchronization pipe in [1].

[1]: https://github.com/opencontainers/runc/blob/v0.0.4/libcontainer/SPEC.md#runtime-and-init-process